lib/update/cluster/steps.go

```diff
@@ -668,6 +669,13 @@ func (r updateStep) etcdPhase(leadMaster storage.Server, otherMasters []storage.
 		restartMasters.AddWithDependency(builder.DependencyForServer(upgradeServers, server), p)
 	}
 
+	// The etcd restart phase resets any etcd clients that may hang watches to the etcd cluster. So ensure the restart
+	// phase is called on each worker to ensure all watches are properly reset.
+	for _, server := range workers {
+		p := r.etcdRestartPhase(server)
+		restartMasters.AddParallel(p)
```

Comment:
Nit: I'd just do `restartMasters.AddParallel(r.etcdRestartPhase(server))` to avoid an extra variable.